src/rt/cpp/vobject.h

```diff
@@ -99,40 +94,38 @@ namespace verona::rt
     }
 
   public:
-    V() : Base(desc()) {}
+    V() : Base() {}
 
     void* operator new(size_t)
     {
       if constexpr (std::is_same_v<Base, Object>)
-        return RegionClass::template create<sizeof(T)>(
-          ThreadAlloc::get(), desc());
+        return RegionClass::template create<desc.size>(
+          ThreadAlloc::get(), &desc);
       else
-        return ThreadAlloc::get()->alloc<sizeof(T)>();
+        return Object::register_object(
+          ThreadAlloc::get()->alloc<desc.size>(), &desc);
     }
 
     void* operator new(size_t, Alloc* alloc)
     {
       if constexpr (std::is_same_v<Base, Object>)
-        return RegionClass::template create<sizeof(T)>(alloc, desc());
+        return RegionClass::template create<desc.size>(alloc, &desc);
       else
-        return alloc->alloc<sizeof(T)>();
+        return Object::register_object(alloc->alloc<desc.size>(), &desc);
     }
 
     void* operator new(size_t, Object* region)
     {
-      if constexpr (std::is_same_v<Base, Object>)
-        return RegionClass::template alloc<sizeof(T)>(
-          ThreadAlloc::get(), region, desc());
-      else
-        return ThreadAlloc::get()->alloc<sizeof(T)>();
+      static_assert(std::is_same_v<Base, Object>, "Don't call on cowns.");
```

Comment:
I don't understand how the expression relates to the message, can you add a comment?  (Or extend the comment at the top of the class to explain the valid values of `Base`)

Reply:
Base is either Cown or Object (not enforced).  This can, and will, be significantly improved for clarity.

Reply:
Can we entirely remove this for non-Object bases? Might require a VBase with the common stuff and a V with two specializations.

Reply:
I have done this now.  Not quite what I think you proposed, but hopefully simpler.